### PR TITLE
Update house model and scene features

### PR DESCRIPTION
**Pull request by abhishekrathi16**
This commit will: 
- [x] Update new files in house.json
- [x] Set up predefined camera views and angles
- [x] Add Skybox for scene

### CODE REVIEW

**Review by chandantiwari1 — COMMENTED** [review]
Code is typesafe.

File: apps/designer/components/Buttons/BECamButton.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      target: {
+        x: 0,
+        y: 700,
+        z: 430,
+      },
+    });
+  }
+  return (
+    <button
+      className="flex justify-center items-center text-xl py-[9px] px-[20px] mr-[10px] duration-300 rounded-[15px] bg-gray-400"
```
Comment (sutyum) [inline apps/designer/components/Buttons/BECamButton.tsx]: CSS Duplications, Why do these button components do not use MenuButton?

File: apps/designer/components/elements/SceneComponent.tsx [hunk 2]
```diff
@@ -35,44 +40,159 @@ const ReactCanvas = ({
...
+
+      //Ground
+      const ground = BABYLON.MeshBuilder.CreateGround("ground", {
+        width: 12000,
+        height: 12000,
+      });
+      ground.position.y = -10;
+      const groundMat = new BABYLON.StandardMaterial("groundMat");
+      groundMat.diffuseTexture = new BABYLON.Texture(
+        "https://assets.vvplus.cc/misc/ground_texture.png"
```
Comment (sutyum) [inline apps/designer/components/elements/SceneComponent.tsx]: refactor base url to a global constant, avoid such duplication

File: apps/designer/package.json [hunk 3]
```diff
@@ -40,10 +42,17 @@
...
         "eslint-config-custom": "workspace:*",
+        "konva": "^8.3.13",
         "next-transpile-modules": "9.0.0",
         "postcss": "^8.4.16",
+        "react-konva": "^18.2.2",
+        "react-native": "^0.70.2",
+        "react-spring": "^9.5.5",
+        "react-zdog": "^1.0.11",
         "tailwindcss": "^3.1.8",
+        "three": "^0.145.0",
```
Comment (sutyum) [inline apps/designer/package.json]: why is this needed?

File: apps/designer/package.json [hunk 4]
```diff
@@ -40,10 +42,17 @@
         "config": "workspace:*",
         "eslint": "7.32.0",
         "eslint-config-custom": "workspace:*",
+        "konva": "^8.3.13",
```
Comment (sutyum) [inline apps/designer/package.json]: why is this needed?

File: apps/designer/package.json [hunk 5]
```diff
@@ -40,10 +42,17 @@
...
         "next-transpile-modules": "9.0.0",
         "postcss": "^8.4.16",
+        "react-konva": "^18.2.2",
+        "react-native": "^0.70.2",
```
Comment (sutyum) [inline apps/designer/package.json]: why is this needed?

File: apps/designer/components/elements/Scene.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import { createCamera } from "./freeCamera";
+import { createGround } from "./ground";
+import { createSkyBox } from "./skybox";
+import { createFog } from "./fog";
+import { createModel } from "./models";
+import * as BABYLON from "@babylonjs/core";
+import { HemisphericLight, Vector3 } from "@babylonjs/core";
+import "@babylonjs/loaders/glTF";
+
+//Work under progress, the demo component contains the code that will use Zod and React Custom Hooks in order to render the scene. This will be a part of the code later on when the work with Zod is complete
```
Comment (sutyum) [inline apps/designer/components/elements/Scene.tsx]: is this needed? Use the format "// TODO: ...." to add semantic comments that can be recognized by code editors

File: apps/designer/components/elements/SceneComponent.tsx [hunk 7]
```diff
@@ -35,44 +40,159 @@ const ReactCanvas = ({
...
+  const navStore = navigationUseStore();
   const minZ = -((config.amount.z * config.separation) / 2);
   useEffect(() => {
     const canvas = canvasRef.current;
     const engine = new Engine(canvas, true);
     const createScene = function () {
       const scene = new Scene(engine);
-      const camera = new ArcRotateCamera(
+
+      //bird's eye view camera
```
Comment (sutyum) [inline apps/designer/components/elements/SceneComponent.tsx]: It would be better to get rid of such comments

File: apps/designer/components/elements/SceneComponent.tsx [hunk 8]
```diff
@@ -35,44 +40,159 @@ const ReactCanvas = ({
...
+        scene
+      );
+
+      freeCamera.wheelPrecision = 1;
+      freeCamera.panningSensibility = 10;
+      freeCamera.lowerRadiusLimit = 500;
+      freeCamera.upperRadiusLimit = 2000;
+      freeCamera.upperBetaLimit = Math.PI / 2;
+
+      // const freeCamera = new ArcRotateCamera(
```
Comment (sutyum) [inline apps/designer/components/elements/SceneComponent.tsx]: It would be better to get rid of such comments

File: apps/designer/components/elements/SceneComponent.tsx [hunk 9]
```diff
@@ -96,35 +216,17 @@ const ReactCanvas = ({
...
-          scene,
-          (newMeshes) => {
-            newMeshes[0].position.y = 0;
-            newMeshes[0].scaling = new Vector3(1, 1, 1);
-            // newMeshes[0].alphaIndex = 0
-          }
-        );
-      }
-      */
+// data[0]?.floors.map((element) => {
```
Comment (sutyum) [inline apps/designer/components/elements/SceneComponent.tsx]: It would be better to get rid of such comments

File: apps/designer/components/Buttons/BECamButton.tsx (hunk 1 above)
Comment (abhishekrathi16) [reply apps/designer/components/Buttons/BECamButton.tsx]: Faced some issues related to functionalities of event listeners, so I created this temporary button. It has been deleted.

File: apps/designer/package.json (hunk 3 above)
Comment (abhishekrathi16) [reply apps/designer/package.json]: Initially we had decided to use React-Spring, so it was a required dependency. Now it has been deleted from the package.json.

File: apps/designer/package.json (hunk 5 above)
Comment (abhishekrathi16) [reply apps/designer/package.json]: Initially we had decided to use React-Spring, so it was a required dependency. Now it has been deleted from the package.json.

**Comment by sutyum** [comment]
You also need to resolve this conflict: apps/designer/pages/index.tsx

File: apps/designer/components/Buttons/BECamButton.tsx (hunk 1 above)
Comment (sutyum) [reply apps/designer/components/Buttons/BECamButton.tsx]: You do not need to delete these, simply add a TODO comment stating pending refactor

**Review by sutyum — CHANGES_REQUESTED** [review]
minor, largely cosmetic changes suggested

**Comment by abhishekrathi16** [comment]
> Conversations awaiting resolution.

Resolved all conversations.

**Review by sutyum — APPROVED** [review]
